app/upgrade/fork.go

```diff
@@ -58,6 +49,15 @@ func CreateUpgradeHandler(mm *module.Manager, configurator module.Configurator,
 		}
 		icamodule.InitModule(ctx, controllerParams, hostParams)
 
+		newVM, err := mm.RunMigrations(ctx, configurator, vm)
```

Comment:
legit, icamodule.InitModule should be called before mm.RunMigrations as described in this [doc](https://github.com/cosmos/ibc-go/blob/031f6a39d5f3f778e18faeae3242b1213d2fe9bd/docs/migrations/v2-to-v3.md?plain=1#L52)

Reply:
thank @catShaark